### ISSUE TITLE
Phalcon\Mvc\Model\Validator\Uniqueness doesn't work with PHQL literals disabled.

### ISSUE THREAD

**Issue by dschissler** [phalcon/cphalcon#10784]
I'm investigating the the params variable that is constructed for the query and how it interacts with the Model::count method

**Comment by dschissler** [comment]
I've determined that it definitely is occurring within the Model::_groupResult method.

**Comment by dschissler** [comment]
@andresgutierrez Can you offer me any advice on how to compile Phalcon quicker?  I don't believe that it is catching anything between builds because it is really taking forever.

**Comment by andresgutierrez** [comment]
@dschissler Do you have the latest version of Zephir?

**Comment by dschissler** [comment]
Recently as of some weeks.

**Comment by andresgutierrez** [comment]
Can you please update to the latest version?

**Comment by dschissler** [comment]
Yes I will do that now.

**Comment by dschissler** [comment]
Master or Development branch?

**Comment by andresgutierrez** [comment]
Master please

**Comment by dschissler** [comment]
The compile/install process is a bit rather strange.  Is it supposed to be version `0.7.1b`?

**Comment by dschissler** [comment]
@andresgutierrez Ok that seems to be caching now at least when no changes are made.

**Comment by dschissler** [comment]
@andresgutierrez I removed the DI being passed into the static count method because it doesn't use it so as to be able to print the data easier.

The parameters being passed into `manager->createBuilder(params)` are:

```
array (
  'conditions' => '[name] = ?0',
  'bind' => 
  array (
    0 => 'Stuff',
  ),
  'bindTypes' => 
  array (
    0 => 2,
  ),
)
```

I'm not sure why this would fail on this with `Literals are disabled in PHQL statements`.  It fails when I use `error_log(query->getSql()["sql"]);` or when I let it get to execute.

**Comment by dschissler** [comment]
What is really strange is that uniqueness validator is working in one case but failing in the second case.

I've changed Phalcon code to fix the issue but I don't understand why it is the way that it is the way that it is.

I changed [this code](https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model.zep#L1104-L1112) to the following:

```
		/**
		 * Builds the columns to query according to the received parameters
		 */
		if fetch distinctColumn, params["distinct"] {
			let columns = functionName . "(DISTINCT " . distinctColumn . ") AS " . alias;
		} else {
			if fetch groupColumns, params["group"] {
				let columns = groupColumns . ", " . functionName . "(" . groupColumn . ") AS " . alias;
			} else {
				// let columns = functionName . "(" . groupColumn . ") AS " . alias;
				if (functionName == "COUNT") {
					let columns = functionName . "(*) AS " . alias;
				} else {
					let columns = functionName . "(" . groupColumn . ") AS " . alias;
				}
			}
		}
```

In one instance the count works with `COUNT(42) ...` but in the other instance it is failing and I can't see what it tried to generate.  I hacked in a specific case for COUNT so that it does the usual `COUNT(*)`.

Anyways the PHQL Literals checker isn't liking the way this is currently being done.

**Comment by dschissler** [comment]
Ok, I slept, cleaned up my code and this is sort of a state of things.

I did some very quick research on mysql aggregate functions that would be called through `Model::_groupResult` and I see that sometimes you want to use `(*)` and other times you want to calculate based upon a column like with `SUM(column_name)`.  Its beyond me as to how this affects other databases.

So for the moment I'm using my hacked code so that I can continue to have PHQL literals disabled but I feel as though these other grouping methods would also sometimes break in just the same way.  I think that it has to do with constants referring somehow to columns getting added in between the parenthesis of these grouping/calcuation functions.

I really don't have time to dig through the AST and so in the breaking case (on the level that I did with the nested EXISTS fix) nor do I have time to make a test.

Maybe one solution for this is to go through a somewhat commonly use base like Vokuro and to give it the treatment like I gave to Webird by binding all queries and disabling literals by default to encourage more developers to use that feature.

**Comment by andresgutierrez** [comment]
@dschissler Could you please provide a small script to reproduce the issue?

**Comment by dschissler** [comment]
I can try within some days.  I don't need to change out model definitions like with the other scripts and so I can do it an easier to follow all inline way this time.  Can you recommend a simple one page bootstrap for getting a test going?

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
